### ISSUE TITLE
Module build failed: Error: Cannot read property 'filter' of undefined

### ISSUE THREAD

**Issue by Akkuma** [amireh/happypack#44]
I'm running this against Webpack 2.1.0-beta7 in watch mode and this is what happens on file change

```
TypeError: Cannot read property 'filter' of undefined
    at Object.HappyLoader (/Users/omni/Documents/git/vault-extension/node_modules/happypack/lib/HappyLoader.js:14:37)
    at applySyncOrAsync (/Users/omni/Documents/git/vault-extension/node_modules/happypack/lib/applyLoaders.js:340:21)
    at apply (/Users/omni/Documents/git/vault-extension/node_modules/happypack/lib/applyLoaders.js:267:5)
    at /Users/omni/Documents/git/vault-extension/node_modules/happypack/lib/applyLoaders.js:130:7
    at applyPitchLoader (/Users/omni/Documents/git/vault-extension/node_modules/happypack/lib/applyLoaders.js:183:14)
    at applyPitchLoader (/Users/omni/Documents/git/vault-extension/node_modules/happypack/lib/applyLoaders.js:191:14)
    at applyPitchLoaders (/Users/omni/Documents/git/vault-extension/node_modules/happypack/lib/applyLoaders.js:221:4)
    at applyLoaders (/Users/omni/Documents/git/vault-extension/node_modules/happypack/lib/applyLoaders.js:115:3)
    at HappyWorker.compile (/Users/omni/Documents/git/vault-extension/node_modules/happypack/lib/HappyWorker.js:32:3)
    at process.accept (/Users/omni/Documents/git/vault-extension/node_modules/happypack/lib/HappyWorkerChannel.js:24:14)
```

**Comment by amireh** [comment]
Honestly, I would rather wait until webpack2 is out of beta and they freeze their APIs (and internal APIs since HappyPack integrates with those as well) before jumping on its support. Many things like the loader `options` variable and the `loaders` properties are changing as well as the resolving APIs so it will take some work.

That aside, I also want HappyPack to work with both versions so there needs to be a way to switch logic where needed based on the version (instead of saying version X of happypack only works with webpack up to version 1.10, etc.) Although that maybe a nightmare in tests, so we'll see.

TL;DR - webpack2 is lower priority for me until it stabilizes, I'll look into its tickets if I find the time but can't promise.

**Comment by Akkuma** [comment]
So there is a quick hacky fix I made locally, which works around the issue, but I haven't checked if this creates a new problem

`  if (!this.options.plugins) return callback(null, sourceCode, sourceMap);`

**Comment by maysale01** [comment]
Seeing this rather frequently on our builds using **webpack 1.13.1** and just running normally (without watch).
